### PR TITLE
Sort by the nearest vax location (Fix #2)

### LINKED ISSUE #2: Sort by location
Under detail_lokasi there is a lat lon. Would be nice if we can ~~sort by lat lon~~ sort by current location to the vax location

### PR DESCRIPTION
**Pull request by agallio**
Add get current location. Still investigating on how to get the nearest vax location.

### CODE REVIEW

**Comment by kelilipan** [comment]
Maybe we can calculate distance between 2 location?
https://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula

If I'm not wrong, the API also return lat&long, maybe we can use this
image

**Comment by tibudiyanto** [comment]
yep. imo, if we got user current location, just sort everything based of that haversine formula. Simplest i can think of is to map the schedule to generate the distance, then sort it.

**Comment by pveyes** [comment]
Minor feedback: if user already grant location before no need to press button again

You can use navigator.permission to check

**Comment by agallio** [comment]
@pveyes 

Currently Safari is not supported. Is there any other way to get users permissions?

image

Or just change the behavior using `useEffect` hook? So users don't have to click the button every single time they want to sort by location?

**Comment by pveyes** [comment]
it's fine to ignore safari, safari user can keep clicking the button 😂

**Comment by agallio** [comment]
Fix #2 

@tibudiyanto 🚀
image

**Comment by tibudiyanto** [comment]
very coool

**Comment by resir014** [comment]
Linked to #2. Also @agallio there are conflicts, please resolve.

**Comment by agallio** [comment]
Done @resir014. 🚀

File: src/components/VaxLocation.tsx [hunk 1]
```diff
@@ -20,7 +21,15 @@ import {
   WrapItem
 } from '@chakra-ui/react';
 
-export default function VaxLocation({ location }) {
+// {!loading && isUserLocationExist && detail_lokasi.length > 0 ? (
+//   <Box bg={distanceBg[colorMode]} w="100%" p={2}>
+//     <Text align="center">JARAK DARI LOKASI ANDA : {detail_lokasi[0].distance} KM</Text>
```
Comment (tibudiyanto) [inline src/components/VaxLocation.tsx]: can you remove the comments?

Comment (agallio) [reply src/components/VaxLocation.tsx]: Done 👍🏻

**Pull request merged**